master/internal/provisioner/provisioner_test.go

```diff
@@ -309,3 +310,49 @@ func TestProvisionerNotProvisionExtraInstances(t *testing.T) {
 	assert.NilError(t, mock.system.StopAndAwaitTermination())
 	assert.DeepEqual(t, mock.cluster.history[len(mock.cluster.history)-1], newMockFuncCall("list"))
 }
+
+func TestProvisionerTerminateUnconnectedInstances(t *testing.T) {
+	setup := &mockConfig{
+		initTime:               time.Now().Add(-time.Hour),
+		maxAgentStartingPeriod: 3 * time.Minute,
+		maxIdleAgentPeriod:     10 * time.Minute,
+		instanceType: TestInstanceType{
+			Name:  "test.instanceType",
+			Slots: 4,
+		},
+		maxInstances: 100,
+		initInstances: []*Instance{
+			{
+				ID:         "instance1",
+				LaunchTime: time.Now().Add(-time.Hour),
+				AgentName:  "agent1",
+				State:      Running,
+			},
+			{
+				ID: "instance2",
+				// This instance should not be terminated because
```

Comment:
non-blocking: can we reword the id to indicate it's starting up, like `startingInstance`?

Reply:
nonblocking: I mean we could save a comment by using a proper name. :)

Reply:
removed